internal/rollup/rollup.config.js

```diff
@@ -0,0 +1,59 @@
+const rollup = require('rollup');
+const nodeResolve = require('rollup-plugin-node-resolve');
+const commonjs = require('rollup-plugin-commonjs');
+const fs = require('fs');
+const path = require('path');
+
+class NormalizePaths {
+  resolveId(importee, importer) {
+    // process.cwd() is the execroot and ends up looking something like /.../2c2a834fcea131eff2d962ffe20e1c87/bazel-sandbox/872535243457386053/execroot/<workspace_name>
+    // from the path to the es6 output is bazel-out/host/bin/<build_file_path>/rollup.runfiles/<workspace_name>/<build_file_path>/bazel-out/host/bin
+    const workspaceName = process.cwd().split('/').pop();
+
+    const firstSegment = importee.split('/')[0]
+    const importerDir = importer ? path.dirname(importer) : ""
+    const buildFileMatch = importerDir.match(`rollup\\.runfiles\\/${workspaceName}/([\\w\\-. \\/]+)\\/bazel\\-out`)
+    const buildFilePath = buildFileMatch ? buildFileMatch[1] : "";
+
+    if (firstSegment === 'bazel-out') {
+      // entry point is a relative path from the execroot
+      return `${process.cwd()}/${importee}.js`;
+    } else if (firstSegment === '.' || firstSegment === '..') {
+      // relative import
+      return `${importerDir}/${importee}.js`;
+    } else if (firstSegment === workspaceName || (buildFilePath && firstSegment === buildFilePath)) {
+      // most imports start with the workspace name but some start with the build file path
+      // need to get down to the base es6 output path
+      const importerSegments = importerDir.split("/")
+      while (importerSegments.length) {
+        const search = importerSegments.slice(-3);
+        if (search.length == 3 &&
+          search[0] === 'bazel-out' &&
+          search[1] === 'host' &&
+          search[2] === 'bin') {
+          break;
+        }
+        importerSegments.pop();
+      }
+      const basePath = importerSegments.join("/")
+      if (firstSegment === workspaceName) {
+        // drop the workspace name from the importee
+        var importPath = importee.split("/");
+        importPath.shift();
+        importPath = importPath.join("/")
+        return `${basePath}/${importPath}.js`;
+      } else {
+        return `${basePath}/${importee}.js`;
```

Comment:
Most imports include the workspace name such as `angular_bazel_example/src/app`. But some don't include the workspace name. The two that don't in angular_bazel_example are in the es6 version of`app.ngfactory.js`:

```
import * as i0 from "@angular/core";
import * as i1 from "src/app";
import * as i2 from "@angular/common";
import * as i3 from "@angular/platform-browser";
import * as i4 from "angular_bazel_example/src/hello-world/hello-world.module";
import * as i5 from "src/hello-world/hello-world.component.ngfactory";
import * as i6 from "angular_bazel_example/src/hello-world/hello-world.component";
```

Reply:
That should be fixed in Angular already:
https://github.com/angular/angular/issues/21022

let's avoid handling the buggy case, we should be able to use latest Angular and always rely on absolute paths starting with the npm package or bazel workspace.

Reply:
Cool. Not needed anymore with angular 5.2.0. Removed the buggy case.